javagen/src/main/java/com/azure/autorest/model/javamodel/JavaClass.java

```diff
@@ -199,6 +204,64 @@ public final void javadocComment(int wordWrapWidth, Consumer<JavaJavadocComment>
         contents.javadocComment(wordWrapWidth, commentAction);
     }
 
+    @Override
+    public final void javadocComment(int wordWrapWidth, Consumer<JavaJavadocComment> commentDescriptionAction, Consumer<JavaJavadocComment> commentTagsAction, boolean withGeneratedMarker) {
+        javadocComment(commentDescriptionAction, commentTagsAction, withGeneratedMarker, true, wordWrapWidth);
+    }
+
+    @Override
+    public final void javadocComment(Consumer<JavaJavadocComment> commentDescriptionAction, Consumer<JavaJavadocComment> commentTagsAction, boolean withGeneratedMarker) {
+        javadocComment(commentDescriptionAction, commentTagsAction, withGeneratedMarker, false, 0);
+    }
+
+    /**
+     * Helper method to add a javadoc comment to the contents.
+     * @param commentDescriptionAction
+     * @param commentTagsAction
+     * @param withGeneratedMarker
+     * @param withWordWrap
+     * @param wordWrapWidth
+     */
+    private void javadocComment(Consumer<JavaJavadocComment> commentDescriptionAction, Consumer<JavaJavadocComment> commentTagsAction, boolean withGeneratedMarker, boolean withWordWrap, int wordWrapWidth) {
+        addExpectedNewLine();
+        if (commentDescriptionAction != null) {
+            contents.javadocCommentStart();
+            if (withGeneratedMarker) {
+                contents.line(GENERATED_JAVADOC_DESC_START_MARKER);
+            }
+            if (withWordWrap) {
+                contents.withWordWrap(wordWrapWidth, () -> commentDescriptionAction.accept(new JavaJavadocComment(contents)));
+            } else {
+                commentDescriptionAction.accept(new JavaJavadocComment(contents));
+            }
+            if (withGeneratedMarker) {
+                contents.line(GENERATED_JAVADOC_DESC_END_MARKER);
+            }
+            if (commentTagsAction == null) { // if no tags action, then end the javadoc comment
+                contents.javadocCommentEnd();
```

Comment:
Should it add GENERATED_JAVADOC_TAG_START_MARKER and GENERATED_JAVADOC_TAG_END_MARKER?

Reply:
updated, and it will add tags marker even it's `commentTagsAction` is null.